### PR TITLE
feat(dropdown): improve accessibility in relation to search dropdown

### PR DESCRIPTION
**Pull request by Inselhopper**
## Description
in case of using the &lt;label&gt; element before a search dropdown to improove accessibility
- id of module is used to add an unique id to this very label 
- a reference to this label is made via the attribut "labelledby"
- see https://www.w3.org/WAI/tutorials/forms/labels/

## Testcase
https://jsfiddle.net/inselhopper/ot5zqvb2/38/

### CODE REVIEW

File: src/definitions/modules/dropdown.js [hunk 1]
```diff
@@ -366,11 +365,22 @@ $.fn.dropdown = function(parameters) {
...
-              $search = $('<input />')
+              if($module.prev('label').length) {
+                $search = $('<input />')
                 .addClass(className.search)
                 .prop('autocomplete', module.is.chrome() ? 'fomantic-search' : 'off')
+                .attr('aria-labelledby',id+'_formLabel')
                 .insertBefore($text)
-              ;
+                ;
+                $module.prev('label').attr('id', id + '_formLabel');
```
Comment (hdogan) [inline src/definitions/modules/dropdown.js:375]: What if I already set `id` attribute to that label?

File: src/definitions/modules/dropdown.js [hunk 2]
```diff
@@ -366,11 +365,22 @@ $.fn.dropdown = function(parameters) {
...
+              }
+              else {
```
Comment (lubber-de) [inline src/definitions/modules/dropdown.js:377]: Let's not duplicate the code here.
I suggest to change your PR into something like this (which will also support an already existing id as [questioned by @hdogan](https://github.com/fomantic/Fomantic-UI/pull/2409/files#r928965797) )
```javascript
var labelNode = $module.prev('label');
$search = $('<input />')
  .addClass(className.search)
  .prop('autocomplete', module.is.chrome() ? 'fomantic-search' : 'off')
;
if (labelNode.length) {
  if(!labelNode.attr('id')) {
    labelNode.attr('id', module.get.id() + '_formLabel');
  }
  $search.attr('aria-labelledby',labelNode.attr('id'));
}
$search.insertBefore($text);
```

**Review by lubber-de — CHANGES_REQUESTED** [review]
Thanks for the PR. Please see my comment for a possible change.

File: src/definitions/modules/dropdown.js [hunk 3]
```diff
@@ -110,8 +110,8 @@ $.fn.dropdown = function(parameters) {
               settings.ignoreDiacritics = false;
               module.error(error.noNormalize, element);
             }
-
-            module.setup.layout();
+            module.create.id();
+            module.setup.layout(id);
```
Comment (lubber-de) [inline src/definitions/modules/dropdown.js:114]: The (id) parameter is not consumed inside the layout function.
As i suggested in https://github.com/fomantic/Fomantic-UI/pull/2409/files#r932673809 the "module.get.id()" function should be used instead, so please remove the provided id as a function call parameter.

File: src/definitions/modules/dropdown.js (hunk 1 above)
Comment (Inselhopper) [reply src/definitions/modules/dropdown.js]: thanks a lot for your objection, it is fixed

File: src/definitions/modules/dropdown.js (hunk 2 above)
Comment (Inselhopper) [reply src/definitions/modules/dropdown.js]: thank you so much, it is a lot better now

**Review by lubber-de — APPROVED** [review]
LGTM

**Comment by lubber-de** [comment]
@all-contributors please add @Inselhopper for code and a11y

**Pull request merged**